core/src/main/scala/shapeless/ops/hlists.scala

```diff
@@ -955,7 +955,36 @@ object hlist {
       }
   }
 
+  /**
+   * Type class supporting access to the first element of this `HList` in `S`.
```

Comment:
I thought about this some more. Note that there are two options:
  1. Select the first element of `L` which is present in `S`
  2. Select the first element of `S` which is present in `L`

The difference is subtle. I read this comment as describing 1. but the implementation as 2.
We should take a step back to make sure that:
  * We implement the semantics that make more sense. Not sure which one, but the user could easily switch by swapping the arguments.
  * The comment and the implementation align.

Do I interpret this correctly?

Reply:
Considering the `selectFirst` syntax I think the implementation (option 2.) makes sense and we need to tweak the comment.